plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -332,6 +334,12 @@ defineBuiltinTerms = do
     -- Text constant
     defineBuiltinTerm annMayInline 'Builtins.emptyString $ PIR.mkConstant annMayInline ("" :: Text)
 
+    -- List constants
+    defineBuiltinTerm annMayInline 'Builtins.mkNilInteger $
```

Comment:
`mkNilInteger` and `mkNilBool` are not builtins, so `defineBuiltinTerm` is both a misnomer and unnecessary. How about `` mkNil :: DefaultUni `Contains` a => Proxy a -> BuiltinList a ``, and handle it in the plugin similar to `&&` and `||`?

Reply:
I'm following the existing convention to a T, you can see that it's the same `defineBuiltinTerm` for the empty bytestring right above, I've only replicated that for the empty lists. 

> How about mkNil :: DefaultUni `Contains` a => Proxy a -> BuiltinList a, and handle it in the plugin similar to && and ||?

(see my other comment)